framework/generated/generated_vulkan_replay_consumer.cpp

```diff
@@ -988,8 +988,11 @@ void VulkanReplayConsumer::Process_vkCreateGraphicsPipelines(
     if (UseAsyncOperations())
     {
         auto task = AsyncCreateGraphicsPipelines(call_info, returnValue, in_device, in_pipelineCache, createInfoCount, pCreateInfos, pAllocator, pPipelines);
```

Comment:
Wouldn't it make more sense to check for the extern sync'd pipeline cache flag here? That way the code doesn't need to call the AsyncCreate<xxx>Pipelines functions in the first place? 

Mainly thinking about how confusing control flow is when you have to call into a function in order to know if you need to skip the function.

Reply:
_Wouldn't it make more sense to check for the extern sync'd pipeline cache flag here?_

honestly, biggest reason for this control-flow was our code-gen. 
having the statement here would require generating it via python-codegen here:
but because it's only relevant for 2-3 cases, 
I chose to put in the `AsyncCreate***Pipelines` overrides, which are not autogenerated.

Reply:
_when you have to call into a function in order to know if you need to skip the function_
what I liked about returning the default lambda:
you can regard as an extra `bool` return type, indicating success.